### PR TITLE
Files

### PR DESCRIPTION
**Pull request by mamazu**
Implementing file upload as a customer option.

### CODE REVIEW

File: src/Entity/CustomerOptions/CustomerOption.php [hunk 1]
```diff
@@ -191,7 +191,7 @@ public function setConfiguration(array $configuration): void
     {
         // Setting the new values
         foreach ($configuration as $key => $value) {
-            $optionKey                                = str_replace('_', '.', $key);
+            $optionKey                                = str_replace(':', '.', $key);
```
Comment (kortwotze) [inline src/Entity/CustomerOptions/CustomerOption.php]: Is this intentional?

File: src/Traits/OrderItemCustomerOptionCapableTrait.php [hunk 2]
```diff
@@ -74,7 +73,7 @@ public function getCustomerOptionConfigurationAsSimpleArray(): array
     public function setUnitPrice(int $unitPrice): void
     {
         $customerOptionAwareUnitPrice = $this->applyConfigurationPrices($unitPrice);
-        $this->unitPrice = $customerOptionAwareUnitPrice;
+        $this->unitPrice              = $customerOptionAwareUnitPrice;
```
Comment (kortwotze) [inline src/Traits/OrderItemCustomerOptionCapableTrait.php]: This needs to be integrated with the SpecialPricingPlugin proper. This comment is just informational.

File: src/Entity/CustomerOptions/CustomerOption.php (hunk 1 above)
Comment (mamazu) [reply src/Entity/CustomerOptions/CustomerOption.php]: Well when using _ this conflicts with the Symfony's block naming convention. This is a bug fix.

File: src/Traits/OrderItemCustomerOptionCapableTrait.php (hunk 2 above)
Comment (mamazu) [reply src/Traits/OrderItemCustomerOptionCapableTrait.php]: This is not part of the pull request.

File: src/Entity/CustomerOptions/CustomerOption.php (hunk 1 above)
Comment (kortwotze) [reply src/Entity/CustomerOptions/CustomerOption.php]: :+1:

File: src/Traits/OrderItemCustomerOptionCapableTrait.php (hunk 2 above)
Comment (kortwotze) [reply src/Traits/OrderItemCustomerOptionCapableTrait.php]: :+1:

**Pull request merged**